### PR TITLE
[stable/rabbitmq] Fix affinity templating

### PR DESCRIPTION
**Pull request by desaintmartin**
Fix affinity templating and add test to prove that it works.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

**Review by alemorcuq — APPROVED** [review]

**Comment by alemorcuq** [comment]
Thanks for the PR, @desaintmartin.

@juan131, @carrodher, please proceed.

File: stable/rabbitmq/templates/statefulset.yaml [hunk 1]
```diff
@@ -44,7 +44,7 @@ spec:
       affinity:
 {{- if .Values.affinity }}
     {{- with .Values.affinity }}
-{{ tpl . $ | indent 8 }}
+{{ tpl (toYaml .) $ | indent 8 }}
```
Comment (juan131) [inline stable/rabbitmq/templates/statefulset.yaml]: We found a lot of issue with bad-formatted YAML(s) when using this combo: `tpl (toYaml .) $`
We workaround it creating a new macro.

Could you please add this macro to the `_helpers.tpl`?

```
{{/*
Renders a value that contains template.
Usage:
{{ include "rabbitmq.tplValue" (dict "value" .Values.path.to.the.Value "context" $) }}
*/}}
{{- define "rabbitmq.tplValue" -}}
    {{- if typeIs "string" .value }}
        {{- tpl .value .context }}
    {{- else }}
        {{- tpl (.value | toYaml) .context }}
    {{- end }}
{{- end -}}
```

Then, use:

```
        {{- if .Values.affinity }}
        affinity: {{- include "rabbitmq.tplValue" (dict "value" .Values.affinity "context" $) | nindent 8 }}
        {{- end }}
```

**Review by juan131 — CHANGES_REQUESTED** [review]

File: stable/rabbitmq/templates/statefulset.yaml (hunk 1 above)
Comment (desaintmartin) [reply stable/rabbitmq/templates/statefulset.yaml]: Indeed. done!

File: stable/rabbitmq/templates/statefulset.yaml [hunk 2]
```diff
@@ -41,12 +41,9 @@ spec:
...
       serviceAccountName: {{ template "rabbitmq.fullname" . }}
       {{- end }}
-      affinity:
-{{- if .Values.affinity }}
-    {{- with .Values.affinity }}
-{{ tpl . $ | indent 8 }}
-    {{- end }}
-{{- end }}
+      {{- if .Values.affinity }}
+      affinity: {{- include "rabbitmq.tplValue" (dict "value" .Values.affinity "context" $) | nindent 8 }}
```
Comment (desaintmartin) [inline stable/rabbitmq/templates/statefulset.yaml]: Should this be current context (`.`) instead of root context (`$`)?

Comment (juan131) [reply stable/rabbitmq/templates/statefulset.yaml]: Yes. It should be (`.`)

Comment (desaintmartin) [reply stable/rabbitmq/templates/statefulset.yaml]: Fixed.

**Review by juan131 — APPROVED** [review]
/ok-to-test
/lgtm

**Pull request merged**